install.sh

```diff
@@ -90,106 +87,121 @@ function curl () {
-    env -u LD_LIBRARY_PATH ${CURL} --ssl-reqd --tlsv1.2 -C - "${@}" && return 0
-    echo_info "Retrying, $((3-i)) retries left."
+    # Force TLS as we know GitLab supports it
+    if [ "$(env -u LD_LIBRARY_PATH "${CURL}" --ssl-reqd --tlsv1.2 -C - "${@}")" ]; then
+      echo_info "Retrying, $((3-i)) retries left."
+    else
+      return 0
+    fi
   done
   # The download failed if we're still here.
   echo_error "Download failed :/ Please check your network settings."
   return 1
 }
 
-case "${ARCH}" in
-"i686"|"x86_64"|"armv7l"|"aarch64")
-  LIB_SUFFIX=
-  # See https://superuser.com/a/1369875
-  # If /bin/bash is 64-bit, then set LIB_SUFFIX, as this is true on both
-  # x86_64 and aarch64 userspace
-  # shellcheck disable=SC2046
-  [ $(od -An -t x1 -j 4 -N 1  /bin/bash) == 02 ] && LIB_SUFFIX='64'
-  if [[ $LIB_SUFFIX == '64' ]] && [[ $ARCH == 'aarch64' ]]; then
-    echo_error "Your device is not supported by Chromebrew yet :/"
-    exit 1
-  fi
-  ;;
-*)
+# See https://superuser.com/a/1369875
+# If /bin/bash is 64-bit, then set LIB_SUFFIX, as this is true on both
+# x86_64 and aarch64 userspace
+# shellcheck disable=SC2046
+[ $(od -An -t x1 -j 4 -N 1  /bin/bash) == 02 ] && LIB_SUFFIX='64'
+if [[ $LIB_SUFFIX == '64' ]] && [[ $ARCH == 'aarch64' ]]; then
   echo_error "Your device is not supported by Chromebrew yet :/"
   exit 1
-  ;;
-esac
+fi
 
-echo_info "\n\nDoing initial setup for install in ${CREW_PREFIX}."
-echo_info "This may take a while if there are preexisting files in ${CREW_PREFIX}...\n"
+# Delete /usr/local/{var,local} symlinks which exist on some Chromium OS distros.
+if [ "${CREW_PREFIX}" = "/usr/local" ]; then
+  [ -L ${CREW_PREFIX}/var ] && sudo rm -f "${CREW_PREFIX}/var"
+  [ -L ${CREW_PREFIX}/local ] && sudo rm -f "${CREW_PREFIX}/local"
+fi
 
-# This will allow things to work without sudo.
-crew_folders="bin cache doc docbook etc include lib lib$LIB_SUFFIX libexec man sbin share tmp var"
-for folder in $crew_folders
-do
-  if [ -d "${CREW_PREFIX}"/"${folder}" ]; then
-    echo_intra "Resetting ownership of ${CREW_PREFIX}/${folder}"
-    sudo chown -R "$(id -u)":"$(id -g)" "${CREW_PREFIX}"/"${folder}"
+if [ -d ${CREW_PREFIX} ]; then
+  if [ "$(ls -A ${CREW_PREFIX})" ]; then
+    echo_error "${CREW_PREFIX} is not empty, would you like it to be cleared?"
+    echo_info "This will delete ALL files in ${CREW_PREFIX}!"
+    echo_info "Continue?"
+    select continue in "Yes" "No"; do
+      if [[ "${continue}" == "Yes" ]]; then
+        sudo rm -rf "${CREW_PREFIX}/*"
+        sudo rm -rf "${CREW_PREFIX}/.*"
+        break 2
+      else
+        exit 1
+      fi
+    done
   fi
-done
+else
+  sudo mkdir ${CREW_PREFIX}
+fi
+
+# This will allow things to work without sudo.
 sudo chown "$(id -u)":"$(id -g)" "${CREW_PREFIX}"
+# This will create the directories.
+crew_folders="bin cache doc docbook include lib/crew/packages lib$LIB_SUFFIX libexec man sbin share var etc/crew/meta etc/env.d tmp/crew/dest"
```

Comment:
Add `etc/bash.d`?

Reply:
`etc/bash.d` is not used in the install process, unlike `etc/env.d` which we export some variables into, and is created by the installation of `crew_profile_base` which is installed when installing `core`.